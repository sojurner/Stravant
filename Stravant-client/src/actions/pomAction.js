export const togglePomState = bool => ({
  type: 'TOGGLE_POM_STATE',
  bool
});

export const setPomHistory = history => ({
  type: 'SET_POM_HISTORY',
  history
});

export const setLastPom = str => ({
  type: 'SET_LAST-POM',
  str
});
