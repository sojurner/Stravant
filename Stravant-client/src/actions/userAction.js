export const setAccessToken = token => ({
  type: 'SET_ACCESS_TOKEN',
  token
});

export const setWeeklyStats = stats => ({
  type: 'SET_WEEKLY_STATS',
  stats
});

export const setTotalStats = stats =>
  console.log(stats) || {
    type: 'SET_TOTAL_STATS',
    stats
  };
