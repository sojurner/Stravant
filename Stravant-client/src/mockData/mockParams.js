export const tokenParams = [
  'https://www.strava.com/oauth/token',
  {
    body:
      '{"client_id":28236,"client_secret":"829f12d9bdb8ce424edf87deff4551c117d3ed36","code":"r5423rffda4c234ctwtwert2ct"}',
    headers: { 'Content-Type': 'application/json' },
    method: 'POST'
  }
];

export const aggregateParams = [
  'https://www.strava.com/api/v3/athletes/28236/stats',
  {
    headers: {
      Authorization: 'Bearer a7eeea216a46a760a74c60acbffc3b55c66537c5',
      'Content-Type': 'application/json'
    },
    method: 'GET'
  }
];

export const weeklyParams = [
  'https://www.strava.com/api/v3/athlete/activities?before=1536123319&after=1536036919',
  {
    headers: {
      Authorization: 'Bearer a7eeea216a46a760a74c60acbffc3b55c66537c5',
      'Content-Type': 'application/json'
    },
    method: 'GET'
  }
];
