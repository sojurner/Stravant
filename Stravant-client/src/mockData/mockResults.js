export const aggregateResult = { Biked: 0, Ran: 6.472964574269732, Swam: 0 };

export const tokenResult = {
  access_token: 'a7eeea216a46a760a74c60acbffc3b55c66537c5',
  athlete: {
    badge_type_id: 0,
    city: null,
    country: null,
    created_at: '2018-09-04T22:53:48Z',
    email: 'pykim0591@gmail.com',
    firstname: 'Paul',
    follower: null,
    friend: null,
    id: 34515396,
    lastname: 'K',
    premium: false,
    profile:
      'https://lh4.googleusercontent.com/-q79bv4Nc_dY/AAAAAAAAAAI/AAAAAAAAAAA/APUIFaMcxPJyTx-ZbjWs8aCsKgkL96lV4w/mo/photo.jpg',
    profile_medium:
      'https://lh4.googleusercontent.com/-q79bv4Nc_dY/AAAAAAAAAAI/AAAAAAAAAAA/APUIFaMcxPJyTx-ZbjWs8aCsKgkL96lV4w/mo/photo.jpg',
    resource_state: 2,
    sex: 'M',
    state: null,
    summit: false,
    updated_at: '2018-09-07T12:20:05Z',
    username: null
  },
  token_type: 'Bearer'
};

export const weeklyResult = {
  Friday: 3784.3,
  Monday: 0,
  Saturday: 0,
  Sunday: 0,
  Thursday: 0,
  Tuesday: 0,
  Wednesday: 0
};
