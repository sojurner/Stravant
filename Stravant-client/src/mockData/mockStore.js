export const currentUser = {
  info: {
    accessToken: 'a7eeea216a46a760a74c60acbffc3b55c66537c5',
    userId: 34515396,
    firstName: 'Paul',
    gender: 'M',
    profilePic:
      'https://lh4.googleusercontent.com/-q79bv4Nc_dY/AAAAAAAAAAI/AAAAAAAAAAA/APUIFaMcxPJyTx-ZbjWs8aCsKgkL96lV4w/mo/photo.jpg'
  },
  totalStats: {},
  weeklyStats: {}
};

export const pomStatus = {
  pomStatus: true,
  lastPom: '',
  pomHistory: {
    'Tue, Sep 11, 2018 8:39 PM': { second: 19, minute: 0, hour: 0 },
    'Tue, Sep 11, 2018 8:40 PM': { second: 3, minute: 0, hour: 0 },
    'Tue, Sep 11, 2018 10:22 PM': { second: 3, minute: 0, hour: 0 }
  }
};

export const clubInfo = {
  clubInfo: { clubName: 'Stravant Club', clubId: 473825 },
  clubActivity: {
    mostRecent: {
      name: 'Pamela',
      distance: 4.270478558110628,
      activity: 'Morning Run'
    },
    Pamela: {
      totalDistance: 62.88999999999999,
      totalTime: 635,
      membersRecent: 'Afternoon Run'
    },
    Paul: { totalDistance: 6.46, totalTime: 78, membersRecent: 'Morning Run' },
    Jesse: {
      totalDistance: 277.78000000000003,
      totalTime: 5292,
      membersRecent: 'Morning Run'
    }
  }
};

export const mockStore = {
  ...currentUser,
  ...pomStatus,
  ...clubInfo
};
