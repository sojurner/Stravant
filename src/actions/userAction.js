export const setAccessToken = token => ({
  type: 'SET_ACCESS_TOKEN',
  token
});

export const togglePomState = bool => ({
  type: 'TOGGLE_POM_STATE',
  bool
});
