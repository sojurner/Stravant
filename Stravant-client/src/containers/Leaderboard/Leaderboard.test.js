import React from 'react';
import {
  mapStateToProps,
  mapDispatchToProps,
  LeaderBoard
} from './Leaderboard';
import { setClubActivity } from '../../actions/clubAction';
import * as store from '../../mockData/mockStore';

describe('Leaderboard', () => {
  let wrapper;
  let mockSetActivity;

  beforeEach(() => {
    mockSetActivity = jest.fn();
    wrapper = shallow(
      <LeaderBoard
      // currentUser={store.mockCurrentUser}
      // clubs={store.clubInfo}
      // setClubActivity={mockSetActivity}
      />
    );
  });

  it('should map the store correctly', () => {
    const mockStore = store.mockStore;
    const expected = { clubs: undefined, currentUser: undefined };
    const mapped = mapStateToProps(mockStore);
    expect(mapped.currentUser).toEqual(mockStore.currentUser);
  });
});
