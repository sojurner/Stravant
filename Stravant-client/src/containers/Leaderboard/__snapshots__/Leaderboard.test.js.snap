// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leaderboard should match snapShot 1`] = `
<div>
  <div>
    <h3
      className="recent-dash"
    >
      Paul did a 1.41
            mile Morning Run
    </h3>
  </div>
  <table
    width="500"
  >
    <tbody>
      <tr>
        <th>
          Name
        </th>
        <th>
          Recent
        </th>
        <th>
          Distance(mi)
        </th>
        <th>
          Duration(m)
        </th>
      </tr>
      <tr>
        <td>
          Jesse
        </td>
        <td>
          Evening Run
        </td>
        <td>
          278.77
        </td>
        <td>
          5297
        </td>
      </tr>
      <tr>
        <td>
          Pamela
        </td>
        <td>
          Afternoon Run
        </td>
        <td>
          62.89
        </td>
        <td>
          635
        </td>
      </tr>
      <tr>
        <td>
          Paul
        </td>
        <td>
          Morning Run
        </td>
        <td>
          7.87
        </td>
        <td>
          101
        </td>
      </tr>
      <tr>
        <td>
          Benjamin
        </td>
        <td>
          Light rail trip
        </td>
        <td>
          1.17
        </td>
        <td>
          20
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Leaderboard should matchSnapshot if there is no club activity 1`] = `
<div>
  <table
    width="500"
  >
    <img
      className="connect-logo"
      height="100"
      src="https://media.giphy.com/media/26xBNwL69IM5ng3f2/source.gif"
      width="100"
    />
    <tbody>
      <tr>
        <th>
          Name
        </th>
        <th>
          Recent
        </th>
        <th>
          Distance(mi)
        </th>
        <th>
          Duration(m)
        </th>
      </tr>
    </tbody>
  </table>
</div>
`;
